src/Client.php

```diff
@@ -324,14 +324,16 @@ public function setUri($uri)
                 $this->clearAuth();
             }
 
+            $uri = $this->getUri();
+
             // Set auth if username and password has been specified in the uri
-            if ($this->getUri()->getUser() && $this->getUri()->getPassword()) {
-                $this->setAuth($this->getUri()->getUser(), $this->getUri()->getPassword());
+            if ($user = $uri->getUser() && $password = $uri->getPassword()) {
+                $this->setAuth($user, $password);
             }
 
             // We have no ports, set the defaults
-            if (! $this->getUri()->getPort()) {
-                $this->getUri()->setPort(($this->getUri()->getScheme() == 'https' ? 443 : 80));
+            if (! $uri->getPort() && $uri->isAbsolute()) {
+                $uri->setPort(($uri->getScheme() == 'https' ? 443 : 80));
```

Comment:
I know it was like that before, but we can do two changes here:
- remove redundant bracket
- change `==` to `===`

Reply:
done ;)